api/src/trace/SugaredTracer.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type {Context, Span, SpanOptions, Tracer} from '../';
+import * as api from '../';
+import {SpanStatusCode} from '../';
+
+const defaultOnException = (e: Error, span: Span) => {
+  span.recordException(e);
+  span.setStatus({
+    code: SpanStatusCode.ERROR
+  });
+};
+
+/**
+ * return a new SugaredTracer created from the supplied one
+ * @param tracer
+ */
+export function wrapTracer(tracer: Tracer): SugaredTracer {
+  return new SugaredTracer(tracer);
+}
+
+class SugaredTracer implements Tracer {
```

Comment:
Bikeshedding: as this is primarily for automatic span end for promises, what about `PromiseTracer`?

Reply:
I understand the main benefit to be just to auto-close spans in general.

Reply:
I think `PromiseTracer` is not fitting as it will not always return a Promise but works with sync functions too. Someone suggested `AutoClosingTracer` ( I think it has been @cartermp ), but this will not fit anymore if we add additional convenience functions.

Reply:
I agree that SugaredTracer is a more general name that allows more options in the future. It also implies, correctly, that it is just a wrapper around a tracer and not anything fundamentally new

Reply:
I would vote +1 in favor of keeping SugaredTracer as the name right now. For some reason I hadn't thought of it as a place to offer more convenience APIs.